lib/elixir/lib/protocol.ex

```diff
@@ -116,6 +116,73 @@ defmodule Protocol do
 
   @doc """
   Derives the `protocol` for `module` with the given options.
+
+  If your implementation passes options or if you are generating
+  custom code based on the struct, you will also need to provide
+  `@fallback_to_any true` or implement a macro defined as
```

Comment:
I believe you don't need to specify `@fallback_to_any` if you want deriving. You can define the any implementation only for deriving without a need to fallback to it. I also wouldn't say `@fallback_to_any` is a derivation. @derive is a compile time choice, @fallback_to_any is a runtime one.

Reply:
This isn't saying you need `@fallback_to_any` for deriving, it's saying you need `@fallback_to_any` if you are generating custom code based on a struct, so it's a subset of cases when you are deriving.

Should there be a separate section for attributes to describe when to use `@derive` vs `@fallback_to_any`?

Reply:
> This isn't saying you need @fallback_to_any for deriving, it's saying you need @fallback_to_any if you are generating custom code based on a struct, so it's a subset of cases when you are deriving.

That's my point. :) You can define the implementation for Any without setting @fallback_to_any to true. Does it make sense? `@fallback_to_any true` means it works without deriving. Without it, it is always opt-in.

Reply:
Got it, I've cleaned up the docs a bit and I think I've got it in a good place